File: src/main/java/ir/reservs/reservs/ui/login/login/LoginContract.java
package ir.reservs.reservs.ui.login.login;

import ir.reservs.reservs.ui.base.BaseFragmentContract;

public class LoginContract {
    interface View extends BaseFragmentContract.View {
        void openMainActivity();

        void showProgress();

        void hideProgress();
    }

    interface presenter extends BaseFragmentContract.Presenter<View> {
        void login(String phone, String password);
    }
}

File: src/main/java/ir/reservs/reservs/ui/login/login/LoginContract.kt
package ir.reservs.reservs.ui.login.login

import ir.reservs.reservs.ui.base.BaseFragmentContract

class LoginContract {
    interface View : BaseFragmentContract.View {
        fun openMainActivity()

        fun showProgress()

        fun hideProgress()
    }

    interface Presenter : BaseFragmentContract.Presenter<View>
}